frontend/vuejs/src/components/TodoList/TodoItem.vue

```diff
@@ -0,0 +1,142 @@
+<script lang="ts">
+import { PropType, ref } from 'vue'
+import { MAX_LEVEL, MIN_LEVEL } from './constants'
+import { ItemList } from '@/types/common'
+import { ARROW_DOWN, ARROW_UP, BACKSPACE, ENTER, TAB } from '@/contants/contants'
+
+export default {
+  props: {
+    item: { type: Object as PropType<ItemList>, required: true },
+    index: Number
+  },
+  expose: ['setFocus'],
+  emits: ['updateItem', 'addNewItem', 'setFocusToElement', 'removeItem'],
+  setup(props, { emit }) {
+    const input = ref(null)
+
+    const setFocus = () => {
+      input.value.focus()
+    }
+
+    const onBlurItemHandler = () => {
+      emit('updateItem', props.item, props.index)
```

Comment:
because I need to know after which index need to add new element.
Do you think that it's better idea to find index in all List, instead to send index?

Reply:
updateItemHandler($event, index)